src/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -505,18 +527,29 @@ private HitsForInsert nextAligned() {
 
     /**
      * Copies alignment info from aligned to unaligned read, clips as appropriate, and sets PG ID.
+     * May also un-map the resulting read if the alignment is bad (e.g. no unclipped bases).
      *
      * @param unaligned Original SAMRecord, and object into which values are copied.
      * @param aligned   Holds alignment info that will be copied into unaligned.
+     * @param isContaminant Should this read be unmapped due to contamination?
      */
-    private void transferAlignmentInfoToFragment(final SAMRecord unaligned, final SAMRecord aligned) {
+    private void transferAlignmentInfoToFragment(final SAMRecord unaligned, final SAMRecord aligned, final boolean isContaminant) {
         setValuesFromAlignment(unaligned, aligned);
         updateCigarForTrimmedOrClippedBases(unaligned, aligned);
         if (SAMUtils.cigarMapsNoBasesToRef(unaligned.getCigar())) {
+            log.warn("Record contains no unclipped bases; making unmapped: " + aligned);
             SAMUtils.makeReadUnmapped(unaligned);
         } else if (SAMUtils.recordMapsEntirelyBeyondEndOfReference(aligned)) {
             log.warn("Record mapped off end of reference; making unmapped: " + aligned);
             SAMUtils.makeReadUnmapped(unaligned);
+        } else if (isContaminant) {
+            log.warn("Record looks like foreign contamination; making unmapped: " + aligned);
+            // NB: for reads that look like contamination, just set unmapped flag and zero MQ but keep other flags as-is.
+            // this maintains the sort order so that downstream analyses can use them for calculating evidence
+            // of contamination vs other causes (e.g. structural variants)
+            unaligned.setReadUnmappedFlag(true);
+            unaligned.setMappingQuality(SAMRecord.NO_MAPPING_QUALITY);
```

Comment:
Because ```SAMUtils.makeReadUnmapped``` takes care of that (plus changing lots of other fields). Whereas ```isContaminant``` is a special case that only affects 0x4 and MQ. I suppose we could make a separate ```SAMUtils``` method for the special case, if that makes things cleaner.